File: app/src/main/java/com/demo/aac/lifecycle/LifecycleActivity.java
package com.demo.aac.lifecycle;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LifecycleRegistry;

public abstract class LifecycleActivity extends Activity implements LifecycleOwner {
    private final LifecycleRegistry lifecycleRegistry = new LifecycleRegistry(this);
    private LifecyclePresenter presenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        presenter = new LifecyclePresenter();
        // 添加生命周期观察者
        getLifecycle().addObserver(presenter);
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        Log.d("gxd", String.format("%s...%s()", this.getClass().getSimpleName(), "onSaveInstanceState"));
        lifecycleRegistry.setCurrentState(Lifecycle.State.CREATED);
        super.onSaveInstanceState(outState);
    }

    @NonNull
    @Override
    public Lifecycle getLifecycle() {
        return lifecycleRegistry;
    }
}

File: app/src/main/java/com/demo/aac/lifecycle/LifecycleActivity.kt
package com.demo.aac.lifecycle

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.view.Gravity
import android.widget.TextView

/**
 * Created by guoxiaodong on 3/26/21 10:51
 */
class LifecycleActivity : AbsLifecycleActivity() {
    private val presenter by lazy {
        LifecyclePresenter(this).also(lifecycle::addObserver)
    }
    private val textView by lazy {
        TextView(this).apply {
            text = "Hello World!"
            textSize = 40f
            gravity = Gravity.CENTER
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(textView)

        presenter.getDataOnStart(this)

        if (requestedOrientation == ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED) {
            window.decorView.postDelayed({
                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
            }, 10000)
        }
    }

    fun onRequestData(result: String) {
        textView.text = result
    }
}